controlplane/src/core/repositories/OrganizationRepository.ts

```diff
@@ -812,8 +816,46 @@ export class OrganizationRepository {
     return result[0];
   }
 
-  public deleteOrganization(organizationID: string) {
-    return this.db.delete(organizations).where(eq(organizations.id, organizationID)).execute();
+  public deleteOrganization(
+    organizationId: string,
+    organizationSlug: string,
+    opts: {
+      keycloakClient: Keycloak;
+      keycloakRealm: string;
+    },
+  ) {
+    return this.db.transaction(async (tx) => {
+      // Cancel subscription
+      const billingRepo = new BillingRepository(tx);
+      const billingService = new BillingService(tx, billingRepo);
+      const oidcRepo = new OidcRepository(tx);
+      const oidcProvider = new OidcProvider();
+
+      const subscription = await billingRepo.getActiveSubscriptionOfOrganization(organizationId);
+      if (subscription) {
+        await billingService.cancelSubscription(organizationId, subscription.id, 'Deleted by api');
```

Comment:
Do we create proper audit logs for everything? e.g. when cancelling the subscription for an org.

Reply:
This is taken care of in the stripe webhook